### PR TITLE
Fixed queue popupReference error

### PR DESCRIPTION
**Pull request by rembiesa**

### CODE REVIEW

**Comment by ghost** [comment]
## Black Duck Security Report

Merging #16119 into developer will not change security risk.

[Click here to see full report](https://copilot.blackducksoftware.com/github/repos/YetiForceCompany/YetiForceCRM/pull-requests/16119)

File: modules/Vtiger/uitypes/ChangesJson.php [hunk 1]
```diff
@@ -23,6 +23,13 @@ public function setValueFromRequest(App\Request $request, Vtiger_Record_Model $r
 			$requestFieldName = $fieldName;
 		}
 		$value = $request->getArray($requestFieldName, 'Text');
+		$moduleModel = Vtiger_Module_Model::getInstance($request->get('module'));
+		foreach ($value['changes'] as $fieldName => $val) {
```
Comment (rskrzypczak) [inline modules/Vtiger/uitypes/ChangesJson.php]: ?

File: modules/Vtiger/uitypes/ChangesJson.php [hunk 2]
```diff
@@ -23,6 +23,13 @@ public function setValueFromRequest(App\Request $request, Vtiger_Record_Model $r
 			$requestFieldName = $fieldName;
 		}
 		$value = $request->getArray($requestFieldName, 'Text');
+		$moduleModel = Vtiger_Module_Model::getInstance($request->get('module'));
```
Comment (rskrzypczak) [inline modules/Vtiger/uitypes/ChangesJson.php]:
```suggestion
		$moduleModel = Vtiger_Module_Model::getInstance($request->getModule());
```

File: modules/Vtiger/uitypes/ChangesJson.php [hunk 3]
```diff
@@ -23,6 +23,13 @@ public function setValueFromRequest(App\Request $request, Vtiger_Record_Model $r
...
+		foreach (array_keys($value['changes']) as $fieldName) {
+			if ($moduleModel->getFieldByName($fieldName)) {
```
Comment (rskrzypczak) [inline modules/Vtiger/uitypes/ChangesJson.php]: ->getFields() : array_kays
$value['changes'] = array_intersect_key($value['changes'], $->getFields());

**Review by rskrzypczak — CHANGES_REQUESTED** [review]

**Pull request merged**